### PR TITLE
Add `Set` metric type to threadstats

### LINKED ISSUE #131: Threadstatsd set metric
There appears to be a disconnect between dogstatd and threadstatsd.

the `set` method is not supported. What is the equivalent in the thread version?

https://github.com/DataDog/datadogpy/blob/master/datadog/dogstatsd/base.py#L227

### PR DESCRIPTION
**Pull request by zippolyte**
### What does this PR do?

Closes #131 

### Description of the Change

Add `Set` class and `set` method to threadstats object to support set metrics in threadstats

### Verification Process

unit tests

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: datadog/threadstats/base.py [hunk 1]
```diff
@@ -285,13 +298,16 @@ def get_user(user_id):
             finally:
                 stats.histogram('user.query.time', time.time() - start)
         """
+
```
Comment (jirikuncar) [inline datadog/threadstats/base.py]: Was it done by an automatic formatter❔

File: datadog/threadstats/base.py [hunk 2]
```diff
@@ -285,13 +298,16 @@ def get_user(user_id):
...
         def wrapper(func):
             @wraps(func)
             def wrapped(*args, **kwargs):
                 with self.timer(metric_name, sample_rate, tags, host):
                     result = func(*args, **kwargs)
                     return result
+
```
Comment (jirikuncar) [inline datadog/threadstats/base.py]: ❔

File: datadog/threadstats/base.py [hunk 3]
```diff
@@ -285,13 +298,16 @@ def get_user(user_id):
...
             return wrapped
+
```
Comment (jirikuncar) [inline datadog/threadstats/base.py]: ❔

File: datadog/threadstats/metrics.py [hunk 4]
```diff
@@ -28,6 +28,25 @@ def flush(self, timestamp, interval):
         raise NotImplementedError()
 
 
+class Set(Metric):
+    """ A gauge metric. """
```
Comment (jirikuncar) [inline datadog/threadstats/metrics.py]:
```suggestion
    """ A set metric. """
```

File: tests/unit/threadstats/test_threadstats.py [hunk 5]
```diff
@@ -25,10 +25,6 @@
 logger.setLevel(logging.ERROR)
 
 
-def assert_equal(a, b):
```
Comment (jirikuncar) [inline tests/unit/threadstats/test_threadstats.py]: 👍

File: datadog/threadstats/base.py (hunk 1 above)
Comment (zippolyte) [reply datadog/threadstats/base.py]: oh yes. It is to separate the inner function definition from the rest of the function code. PyCharm formatted that automatically

**Pull request merged**